spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -118,8 +118,7 @@ public class SparkCatalog extends BaseCatalog {
    */
   protected Catalog buildIcebergCatalog(String name, CaseInsensitiveStringMap options) {
     Configuration conf = SparkUtil.hadoopConfCatalogOverrides(SparkSession.active(), name);
-    Map<String, String> optionsMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
-    optionsMap.putAll(options);
+    Map<String, String> optionsMap = new TreeMap<>(options.asCaseSensitiveMap());
```

Comment:
@rajarshisarkar, don't we still need to create the map with `CASE_INSENSITIVE_ORDER`? We want to use the original keys, but it should not force case sensitivity. So this should be:

```java
Map<String, String> optionsMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
optionsMap.putAll(options.asCaseSensitiveMap());
```

Reply:
I see, I have made the change.